### DISCUSSION TITLE
Skip many waypoints at once

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#17] [Q&A] [answered]
at the moment short press on directional arrow lop left shows the navigation submenu where you can skip waypoints.
_short press on the street info bar has no function_
long press on the street info bar shows up the main waypoints, when selecting them distance and time is displayed in the top bar. So far so nice.

just a proposal: Could we make short press on the street info bar showing the complete list of SPs and WPs which have not yet been passed? And then offer the possibility to swipe one or more SPs to the right in order to delete them from the list? After swiping for example the next SP, directly update the list by moving the next SP to the top. When pressing again short on the street info bar make a recalculation to the next (not deleted) SP/WP?

**Comment by devemux86** [comment]
Thanks for the interesting suggestion, this should be easy!

---

Note that the top panels are optional, also about their events:

Long press on the "next waypoint" panel shows the next waypoints selection list.
Tap on the "next waypoint" panel centers the map (temporarily) on the waypoint.

Tap on the "turn instruction" panel speaks the next turn instruction.

---

We can extend the existing actions dialog (tap on top-left direction icon):
- Skip next waypoint
- **Skip next waypoints** (remove or keep "Skip next waypoint"?)
- Avoid roadblock

The "Skip next waypoints" can show the next waypoints list.
Select a waypoint and it will recalculate from the user location to that waypoint.

How this workflow looks like (with fewer clicks)?

**Reply by DolomitesTom** [reply]
marvellous idea, but I would like to keep the skip next waypoint menu for just skipping one shaping point. Please also enhance the spacing between the submemu entries in order to be able to easily tapping with thick motorcycle gloves (i also drive in winter season as long as no ice is on the road)

**Reply by devemux86** [reply]
I have already put it in my list for implementation. :slightly_smiling_face:

**Comment by devemux86** [comment]
**Skip many waypoints** at once will be available in the next Cruiser update.

Again with a **partial re-routing**, the rest of the **route remains unchanged**.
This waypoint selection list contains both via + shaping points.

<img src="https://user-images.githubusercontent.com/3484020/217534319-90c30636-b0a7-42fc-8d4c-df16fb554eaa.jpg" width="50%"/><img src="https://user-images.githubusercontent.com/3484020/217534336-763c9709-c173-4dc7-8b08-2631170d0738.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
cool... is it possible to mark several waypoints at once by tap, hold and move downwards (as an alternative one action movement instead of tap, tap, tap...)... is it clear what I mean (am not a native english speaker)?

**Reply by devemux86** [reply]
The waypoint list shows all the following waypoints:
- From next waypoint (shaping point 3 in above example, see map in left image)
- All the way to the destination ("Krusovice" in above example)

You can select any waypoint in the list and the partial re-routing will skip the ones before it.
Essentially you **select the next intermediate destination**, where it will route and navigate.

**Reply by devemux86** [reply]
The window is similar to the waypoints list that appears when you long press the waypoint panel
at the top, to show the distance + name of a next via point. But that list _only_ shows the via points.
("185 km · Pekelne Doly" in above example)

However with "Skip waypoints", the waypoints selection list contains both **via + shaping points**.

**Reply by DolomitesTom** [reply]
Even better, forgot what was planned for a moment 🙈

**Comment by Ciutnik** [comment]
Clever! 👍 
And by the way, [Pekelné Doly](https://maps.app.goo.gl/FSUQGkuo6zYprD2C9) is my dream via-point of my future trip... which unfortunately will probably never happen. I can only regret that I have not visited this interesting place many times being close in the past.

**Comment by devemux86** [comment, marked as answer]
Skip many waypoints at once in navigation is available in [Cruiser 3.0.9](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

- Tap top-left navigation panel
Details in https://github.com/devemux86/cruiser/discussions/17#discussioncomment-4904776)

**Comment by DolomitesTom** [comment]
Works perfect

**Comment by devemux86** [comment]
https://github.com/devemux86/cruiser/discussions/57#discussioncomment-5215815 by @Leeb64

> At Skip ? : When skipping waypoints, the numbering is incorrect. Skip WP1->Nothing Happens Skip WP2-> WP1 is skipped etc.

The waypoint list shows _all the next waypoints_ and we **select the target waypoint**
to perform a new route calculation, skipping the ones in between.

Maybe we can use a different title on the dialog?

**Reply by DolomitesTom** [reply]
maybe
Skip 1 WP
Skip X WPs

**Reply by devemux86** [reply]
I meant the title of the dialog with waypoints list, after press the skip button.
Instead of the existing title "Waypoint", change it to "Destination" or "Target".

**Reply by DolomitesTom** [reply]
yes, better destination, or next destination.

**Comment by Leeb64** [comment]
Bei Skip ? :
Beim Überspringen von Wegpunkten ist die Nummerierung falsch.
WP1 überspringen -> Nichts passiert
WP2 überspringen -> WP1 wird übersprungen usw.

**Reply by Leeb64** [reply]
OKAY ....das ganze falsch verstanden...

**Reply by devemux86** [reply]
In this dialog we select the **target waypoint** to perform route calculation.

The next waypoint is already the next, it cannot skip anything.
Selecting the 2nd next waypoint skips the 1st, etc.

---

Maybe we need to display it better, see my comment above: https://github.com/devemux86/cruiser/discussions/17#discussioncomment-5215938

**Reply by Leeb64** [reply]
I thought that the WP to be skipped are selected here.

**Comment by devemux86** [comment]
In the next Cruiser update, the waypoints list will have the title "Destination"
and the list will contain the next waypoints (except for the next one).

So it should be more clear?

**Reply by DolomitesTom** [reply]
yes

**Comment by DolomitesTom** [comment]
is this new redesign look also planned for the skip several wps function? I don't know any more where I had made the proposal to show only WPs (not SPs) in own squares for use with 1 fingertip and auto-ok.

**Reply by devemux86** [reply]
It is already in my list, it is just more work, so will need more time. :slightly_smiling_face:

**Reply by devemux86** [reply]
We had discussed it here:
- https://github.com/devemux86/cruiser/discussions/82#discussioncomment-5343132

**Reply by DolomitesTom** [reply]
OK perfect, no need to hurry :-)

**Comment by devemux86** [comment]
Why also showing SPs? Because of the missing curvy routing option, you normally need to set more shaping points in order to use the preferred streets. Therefore shaping points have in princple no meaning. During navigation you can't see how many SPs have to be deleted in order to use a different road. But WPs you normally have in mind. Therefore I find it better to only show WPs as extra buttons for skipping all SPs between GPS position and that WP.

_Originally posted by @DolomitesTom in https://github.com/devemux86/cruiser/discussions/82#discussioncomment-5344854_

**Reply by WolfgangZ42** [reply]
> Because of the missing curvy routing option, you normally need to set more shaping points in order to use the preferred streets.

But we also need the SPs to not lose too much of the curvy route through BRouter in case of detours!
Probably I will not use "skip many waypoints at once" anyway and hope that I can skip individual SPs.

**Reply by devemux86** [reply]
"Skip next waypoint" works with both via and shaping points.

What we need in "Skip many waypoints"?
Via and shaping points or only via points?

**Comment by DolomitesTom** [comment]
The problem is maybe not if showing only via points or via and shaping points in an option list. Scrolling through a list of many SPs/WPs quickly becomes confusing and then finding the right option and pressing OK again is much too cumbersome.

Normally you do this, I would say, if you want to shorten the way to the intermediate destination, so consciously want to say goodbye to the curvy route till the next via point. I would therefore only list the waypoints (via points) in a rectangle (like the action sub-menu itself, I don't know if this can be built dynamically) with kilometer information from the current position (how far it is to this point (if possible)). Then you have a much better feeling if you can skip the SPs to that destination. Maybe this is technically not possible because the action menu is static. If it can be dynamic and if the route contains more than 8 via points (normally not), then switch to the next 8 rectangles (better usable with gloves than an option list entry). All buttons with Auto-OK (skip Sps to this interim destination). If not otherwise possible, an option list will do as well.

If you get lost, you have to stop anyway to reload the route.
If you just want to skip some SPs then you can use the "skip next waypoint" function several times in sequence.

**Reply by devemux86** [reply]
> Maybe this is technically not possible because the action menu is static. If it can be dynamic

I created the action menu to be auto-generated and scrollable. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
> I created the action menu to be auto-generated and scrollable. 🙂

Then it sounds feasible to show the waypoints in the action buttons. I find the Cancel and OK buttons disturbing, because you have to be careful where to start scrolling and often tap on the ok accidentially if you want to scroll.

Can we make it like this?
Show all shaping points only with the numbers in white color font and via points in yellow with the kilometer info in action button rectangles instead of option list?
Then use 1-tap for selection, swipe to the left for scrolling right, swipe to the right for scrolling left and long-press for Auto-OK? Additionally as Auto-OK: when selecting the same rectangle that has the focus again, then make auto-OK, if selecting a different rectancle then change focus to this item.

**Comment by devemux86** [comment]
Skip many waypoints with the new action menu:

<img src="https://user-images.githubusercontent.com/3484020/229498973-dae2c175-e24e-4c70-97e6-6c272a6bac06.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
Wow, absolute sensational

**Reply by devemux86** [reply]
Select the next waypoint to display distance / name with the new action menu:
(long press top or right distance panel)

<img src="https://user-images.githubusercontent.com/3484020/229499342-4cd19867-5b28-4117-8648-ecac659a7b56.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
Perfect analogy, consistently implemented 👍

**Comment by devemux86** [comment]
> Skip many waypoints with the new action menu

> Select the next waypoint to display distance / name with the new action menu

Both are available in [Cruiser 3.0.16](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by DolomitesTom** [reply]
the biker heart is happy... works

**Reply by WolfgangZ42** [reply]
I was quite skeptical and now have to admit that it does have charm:

Before: ___________________________After:
<img width="200" src="https://user-images.githubusercontent.com/121806238/229812220-1cd867bd-1425-49b1-b48d-a7f5548ae613.png"/>  <img width="200" src="https://user-images.githubusercontent.com/121806238/229812256-7d0634fc-7081-434e-bccd-4092bb47f6fc.png"/>   
Sorry for doubting ...

But in landscape mode I get this:
<img width="400" src="https://user-images.githubusercontent.com/121806238/229814182-d8b16d97-ef4a-469c-b1c9-5354aac5c107.png"/> 
Surely a small thing ...

**Reply by devemux86** [reply]
> But in landscape mode I get this

@WolfgangZ42 thanks for the report! How many waypoints?

It seems I forgot to enable the scroll in this auto-generated menu in landscape mode,
so all are packed in the available space! :slightly_smiling_face: 

It will work properly in the next Cruiser update.

**Reply by WolfgangZ42** [reply]
> How many waypoints?

7 ViaPoints (including Start and End) and ~110 ShapingPoints (automatically set by GPXshaping)

**Reply by devemux86** [reply]
> 110 ShapingPoints

!! Can you please share this route for testing? (or via email)

**Reply by WolfgangZ42** [reply]
Sure: [SaisonStart_2023SHP.zip](https://github.com/devemux86/cruiser/files/11150038/SaisonStart_2023SHP.zip)

**Reply by devemux86** [reply]
@WolfgangZ42 thanks!

After the fix it should look like this:

<img src="https://user-images.githubusercontent.com/3484020/229837743-e650966e-bd5d-4659-8e55-119c73cb43d7.jpg" width="40%"/><img src="https://user-images.githubusercontent.com/3484020/229837762-3f923585-8d66-4576-bfd7-f86f420df598.jpg" width="60%"/>

**Reply by devemux86** [reply]
@WolfgangZ42 thanks again for the report!

It should work properly in [Cruiser 3.0.17](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by DolomitesTom** [reply]
IT works 👍
